ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -99,13 +99,20 @@ private void SetupNotificationSiteMap(IEnumerable<NotificationSite> notification
             ValidateYearComparisonOnModel(ClinicalDetails, nameof(ClinicalDetails.BCGVaccinationYear),
                 ClinicalDetails.BCGVaccinationYear, PatientBirthYear);
             
-            if (!ModelState.IsValid)
+            ClinicalDetails.SetFullValidation(Notification.NotificationStatus);
+
+            var notificationSites = CreateNotificationSitesFromModel(Notification.NotificationId);
+            Notification.NotificationSites = notificationSites.ToList();
+            if (!TryValidateModel(this))
             {
                 return false;
             }
 
+            // Setting NotificationSite to null is required to prevent db updating notificationSites in UpdateClinicalDetailsAsync method
+            Notification.NotificationSites = null;
```

Comment:
I don't like this, but had to do it to avoid saving notification sites (and getting db error) on next line

Reply:
Uh, yeah, that's not great...
Would it be better to create a fresh `Notification` for the purposes of validating notificationSites? Then that would a good place to explain why, too